src/plugins/data_view_field_editor/public/components/field_editor/form_fields/script_field.tsx

```diff
@@ -198,12 +175,18 @@ const ScriptFieldComponent = ({ existingConcreteFields, links, placeholder }: Pr
   }, [error, displayPainlessScriptErrorInMonaco, updateMonacoMarkers]);
 
   useEffect(() => {
+    const sub = PainlessLang.validation$().subscribe(({ isValid, isValidating, errors }) => {
+      controller.setScriptEditorValidation({
+        isValid,
+        isValidating,
+        message: errors[0]?.message ?? null,
+      });
```

Comment:
just quickly tested it, but it seems the issue hasn't been resolved in my local testing, when I removed the `controller.setScriptEditorValidation` code, I could remove the last character of the Painless script


https://user-images.githubusercontent.com/463851/232485178-54d6d16e-3a27-490a-9e25-30c7e38b417c.mp4